lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -1694,23 +1694,51 @@ static LogicalResult verifyPortSymbolUses(FModuleLike module,
   return success();
 }
 
+static LogicalResult verifyEnabledLayers(FModuleLike module,
+                                         SymbolTableCollection &symbolTable) {
+  auto circuitOp = module->getParentOfType<CircuitOp>();
+  for (auto layer : module.getLayers()) {
+    auto *op =
+        symbolTable.lookupSymbolIn(circuitOp, cast<SymbolRefAttr>(layer));
+    if (!op)
+      return module->emitOpError() << "enables unknown layer '" << layer << "'";
+    if (!isa<LayerOp>(op))
```

Comment:
This only checks the final reference is a layer, but structural requirements on layers ensure all parents will also be layerop's.